addons/mail/static/tests/chatter/web/follow_button.test.js

```diff
@@ -5,74 +5,23 @@ import {
 import { describe, test } from "@odoo/hoot";
-import { serverState } from "@web/../tests/web_test_helpers";
 
 describe.current.tags("desktop");
 defineMailModels();
 
-test("base rendering not editable", async () => {
-    await start();
-    await openFormView("res.partner", serverState.partnerId);
-    await contains(".o-mail-Chatter-follow", { text: "Follow" });
-});
-
-test("hover following button", async () => {
-    const pyEnv = await startServer();
-    const threadId = pyEnv["res.partner"].create({});
-    pyEnv["mail.followers"].create({
-        is_active: true,
-        partner_id: serverState.partnerId,
-        res_id: threadId,
-        res_model: "res.partner",
-    });
-    await start();
-    await openFormView("res.partner", threadId);
-    await contains(".o-mail-Chatter-follow", { text: "Following" });
-    // ensure no re-rendering after mouseenter because messages can be loaded before or after data
-    await contains(".o-mail-Thread", { text: "The conversation is empty." });
-    await triggerEvents(".o-mail-Chatter-follow", ["mouseenter"], { text: "Following" });
-    await contains(".o-mail-Chatter-follow", { text: "Unfollow" });
-});
-
-test('click on "follow" button', async () => {
-    await start();
-    await openFormView("res.partner", serverState.partnerId);
-    await contains("button", { text: "Follow" });
-    await click("button", { text: "Follow" });
-    await contains("button", { text: "Following" });
-});
-
-test('Click on "follow" button should save draft record', async () => {
-    await start();
-    await openFormView("res.partner", undefined, {
-        arch: `
-            <form string="Partners">
-                <sheet>
-                    <field name="name" required="1"/>
-                </sheet>
-                <chatter/>
-            </form>`,
-    });
-    await contains("button", { text: "Follow" });
-    await contains("div.o_field_char");
-    await click("button", { text: "Follow" });
-    await contains("div.o_field_invalid");
-});
-
-test('click on "unfollow" button', async () => {
+test("base rendering follow and unfollow button", async () => {
     const pyEnv = await startServer();
     const threadId = pyEnv["res.partner"].create({});
-    pyEnv["mail.followers"].create({
-        is_active: true,
-        partner_id: serverState.partnerId,
-        res_id: threadId,
-        res_model: "res.partner",
-    });
     await start();
     await openFormView("res.partner", threadId);
-    await click(".o-mail-Chatter-follow", { text: "Following" });
-    await contains("button", { text: "Follow" });
+    await contains(".o-mail-Followers-counter", { text: "0" });
+    await click(".o-mail-Followers-button");
+    await click(".o-dropdown-item", { text: "Follow"});
+    await contains(".o-mail-Followers-counter", { text: "1" });
+    await click(".o-mail-Followers-button");
+    await contains(".o-mail-FollowerList-unfollowBtn + .o-mail-Follower-action");
```

Comment:
Avoid asserting classnames that feel too crafted by hand like these. `o-mail-Follower(s)` is fine but here prefer targeting things that end-users actually see.

They see `Unfollow` text and `.fa-pencil[title='Edit subscription']`, prefer these over the assertion above.

I would also omit `+`, this is too bias in exact templating, which doesn't necessarily relate to how this is shown on UI, i.e. this doesn't necessarily mean they are next to each other on the same line. And really this is not that useful so I would drop it because this is too much irrelevant implementation detail.